api/v1/typha_deployment_types.go

```diff
@@ -128,6 +140,11 @@ type TyphaDeploymentSpec struct {
 	// Template describes the typha Deployment pod that will be created.
 	// +optional
 	Template *TyphaDeploymentPodTemplateSpec `json:"template,omitempty"`
+
+	// The deployment strategy to use to replace existing pods with new ones.
+	// +optional
+	// +patchStrategy=retainKeys
+	Strategy *appsv1.DeploymentStrategy `json:"strategy,omitempty" patchStrategy:"retainKeys" protobuf:"bytes,4,opt,name=strategy"`
```

Comment:
I think this means we're allowing Recreate as a strategy, which I don't see any use case for. Possibly allowing a case where typhas can't come up successfully and resulting in outage because all typhas were killed before creating new ones.
I think our intention here was to allow setting MaxUnavailable and MaxSurge on the Rolling updates, so I think we should limit to the RollingUpdate type.